addons/misra.py

```diff
@@ -2019,6 +2019,35 @@ def misra_15_3(self, data):
                             self.reportError(token, 15, 3)
                             break
                         t = t.next
+    
+    def misra_15_4(self, data):
+        # Return a list of scopes affected by a break or goto
+        def getLoopsAffectedByBreak(knownLoops, scope, isGoto):
+            if scope and scope.type and scope.type not in ['Global', 'Function']:
```

Comment:
I would probably have used `if scope and scope.isExecutable:` but it does not really matter.